howso/derive_utilities.amlg

```diff
@@ -14,89 +14,140 @@
 	(declare
 		(assoc derivation_failed (false))
 
-		(map
-			(lambda (let
-				(assoc
-					feature (current_value 1)
-					valid_derivation
-						(call !DeriveSeriesValue (assoc
-							derived_feature (current_value 2)
-							feature_index_map feature_index_map
-							last_series_index last_series_index
-						))
-				)
+		;first pass attempt to derive all the features, create a list of features that did not derive to a valid value (i.e., nulls)
+		(declare (assoc
+			invalid_derivation_features
+				(filter (map
+					(lambda (let
+						(assoc
+							feature (current_value 1)
+							valid_derivation
+								(call !DeriveSeriesValue (assoc
+									derived_feature (current_value 2)
+									feature_index_map feature_index_map
+									last_series_index last_series_index
+								))
+						)
 
-				;feature needs to be generated if it references a row that's not in the current series_data or is out of bounds
-				(if (not valid_derivation)
-					(seq
-							;valid_derivation of (null) means a time feature generated a value past the max bondary, stop the series
-						(if (= (null) valid_derivation)
+						;feature needs to be generated if it references a row that's not in the current series_data or is out of bounds
+						(if (not valid_derivation)
 							(seq
-								(assign (assoc derivation_failed (true)))
-								(conclude)
+								;valid_derivation of (null) means a time feature generated a value past the max bondary, stop the series
+								(if (= (null) valid_derivation)
+									(assign (assoc derivation_failed (true) ))
+								)
+								;else output feature name that failed to derive so its value can be generated below
```

Comment:
the "else" in this comment seems a bit dubious as it's not on an else branch, but should always be output when it was not a valid derivation